src/element.ts

```diff
@@ -8,52 +11,342 @@ export class Element {
   /**
    * The css selector for the element
    */
-  public selector: string; // eg "#user" or "div > #name" or "//h1"
+  readonly #selector: string; // eg "#user" or "div > #name" or "//h1"
 
   /**
    * How we select the element
    */
-  public method: "document.querySelector" | "$x";
+  readonly #method: "document.querySelector" | "$x";
 
   /**
    * The page this element belongs to
    */
-  private page: Page;
+  readonly #page: Page;
+
+  /**
+   * Protocol to use, attached to the page
+   */
+  readonly #protocol: Protocol;
+
+  /**
+   * ObjectId belonging to this element
+   */
+  readonly #objectId?: string;
 
+  /**
+   * @param method - The method we use for query selecting
+   * @param selector - The CSS selector
+   * @param page - The page this element belongs to
+   * @param protocol - The protocol for the page this element belongs to
+   * @param objectId - The object id assigned to the element
+   */
   constructor(
     method: "document.querySelector" | "$x",
     selector: string,
     page: Page,
+    protocol: Protocol,
+    objectId?: string,
   ) {
-    this.page = page;
-    this.selector = selector;
-    this.method = method;
+    this.#objectId = objectId;
+    this.#page = page;
+    this.#selector = selector;
+    this.#method = method;
+    this.#protocol = protocol;
   }
 
   /**
    * Get the value of this element, or set the value
    *
    * @param newValue - If not passed, will return the value, else will set the value
    *
-   * @returns The value if setting, else void if not
+   * @returns The value if setting, else an empty string if not
```

Comment:
says we return an empty string, but the description says this method is to get the value of this element. is it expected that we return an empty string or should we be returning the value of the element if `newValue` is not passed in?